src/ui/molecules/header/Header.tsx

```diff
@@ -1,11 +1,30 @@
-import React, { useCallback, useEffect, useState } from 'react'
-import type { DeepReadonly, UseState } from 'superTypes'
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import classNames from 'classnames'
+import short from 'short-uuid'
+import type { DeepReadonly, UseState } from 'superTypes'
 import { useMediaType } from 'hook/useMediaType'
+import { useOnKeyboard } from 'hook/useOnKeyboard'
+import { useOnClickOutside } from 'hook/useOnClickOutside'
 import { ThemeSwitcher as Switcher } from 'ui/atoms/theme/ThemeSwitcher'
+import { Typography } from 'ui/atoms/typography/Typography'
 import { Icon } from 'ui/atoms/icon/Icon'
 import './header.scss'
 
+type NavigationItemsWithId = DeepReadonly<Map<string, NavigationItem>>
+
+export type NavigationMenuProps = {
+  navigation: NavigationItemsWithId
+  onNavItemSelection: (id: string, isSubItem?: boolean) => void
```

Comment:
It will be better to create an alias for id
```ts
type Id = string
```

Reply:
Good idea!